sharktuner/boo_tuner/boo_tuner.py

```diff
@@ -0,0 +1,410 @@
+# Copyright 2025 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import argparse
+import logging
+import os
+import shlex
+import shutil
+import subprocess
+import sys
+import tempfile
+import traceback
+from collections.abc import Callable
+from datetime import datetime
+from pathlib import Path
+from typing import Optional
+from typing_extensions import override
+
+from sharktuner import common, libtuner
+
+
+class BooPathConfig(libtuner.PathConfig):
+    """Path configuration for BOO tuner with BOO-specific directory naming."""
+
+    def _name_base_dir(self) -> Path:
+        timestamp = datetime.now().strftime("%Y_%m_%d_%H_%M")
+        base_dir = Path(f"./boo_tuning_{timestamp}")
+        return base_dir
+
+    def create_benchmark_path_config(self, benchmark_name: str) -> libtuner.PathConfig:
+        """Create a PathConfig for a specific benchmark under the main BOO tuning directory."""
+        base_dir = self.base_dir
+
+        class BenchmarkPathConfig(libtuner.PathConfig):
+            def _name_base_dir(self) -> Path:
+                return base_dir / benchmark_name
+
+        return BenchmarkPathConfig()
+
+
+class BooTuner(libtuner.TuningClient):
+    """Tuning client for IREE Turbine's BOO (Bag of Ops) kernels."""
+
+    def __init__(self, tuner_context: common.TunerContext):
+        super().__init__(tuner_context)
+        self.compile_flags: list[str] = []
+        self.benchmark_flags: list[str] = []
+        self.compile_timeout: Optional[float] = 16
+        self.benchmark_timeout: Optional[float] = None
+        self.auto_benchmark_timeout: bool = True
+
+    @override
+    def get_iree_compile_flags(self) -> list[str]:
+        return self.compile_flags
+
+    @override
+    def get_iree_compile_timeout_s(self) -> Optional[float]:
+        return self.compile_timeout
+
+    @override
+    def get_iree_benchmark_module_flags(self) -> list[str]:
+        return self.benchmark_flags
+
+    @override
+    def get_iree_benchmark_timeout_s(self) -> Optional[float]:
+        return self.benchmark_timeout
+
+    @override
+    def is_auto_iree_benchmark_timeout(self) -> bool:
+        return self.auto_benchmark_timeout
+
+    @override
+    def should_prune_slower_candidates(self) -> bool:
+        # BooTuner has only one phase, so prune candidates if all are slower than baseline.
+        return True
+
+
+def insert_placeholder_input_file(argv: list[str]) -> list[str]:
+    """Insert a placeholder input file for libtuner compatibility."""
+    return [argv[0], "boo.mlir"] + argv[1:]
+
+
+def arg_parse() -> tuple[argparse.Namespace, list[str]]:
```

Comment:
```suggestion
def parse_args() -> tuple[argparse.Namespace, list[str]]:
```

We start function names with action verbs